mixer/test/spybackend/nosession_integration_test.go

```diff
@@ -17,21 +17,181 @@ package spybackend
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"testing"
 
 	"google.golang.org/grpc"
 
 	"istio.io/api/mixer/adapter/model/v1beta1"
+	istio_mixer_v1 "istio.io/api/mixer/v1"
 	adapter_integration "istio.io/istio/mixer/pkg/adapter/test"
 	"istio.io/istio/mixer/template/listentry"
 	"istio.io/istio/mixer/template/metric"
 	"istio.io/istio/mixer/template/quota"
 )
 
-// This test for now just validates the backend can be started and tested against. This is will be used to verify
-// the OOP adapter work. As various features start lighting up, this test will grow.
+const (
+	h1 = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: handler
+metadata:
+  name: h1
+  namespace: istio-system
+spec:
+  adapter: spybackend-nosession
+  connection:
+    address: "%s"
+---
+`
+	i1Metric = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: i1metric
+  namespace: istio-system
+spec:
+  template: metric
+  params:
+    value: request.size | 123
+    dimensions:
+      destination_service: "\"myservice\""
+      response_code: "200"
+---
+`
+
+	r1H1I1Metric = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: r1
+  namespace: istio-system
+spec:
+  actions:
+  - handler: h1.istio-system
+    instances:
+    - i1metric
+---
+`
+
+	h2 = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: handler
+metadata:
+  name: h2
+  namespace: istio-system
+spec:
+  adapter: spybackend-nosession
+  connection:
+    address: "%s"
+---
+`
+
+	i2Metric = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: i2metric
+  namespace: istio-system
+spec:
+  template: metric
+  params:
+    value: request.size | 456
+    dimensions:
+      destination_service: "\"myservice2\""
+      response_code: "400"
+---
+`
+
+	r2H2I2Metric = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: r2
+  namespace: istio-system
+spec:
+  actions:
+  - handler: h2.istio-system
+    instances:
+    - i2metric
+---
+`
+	i3List = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: i3list
+  namespace: istio-system
+spec:
+  template: listentry
+  params:
+    value: source.name | ""
+---
+`
+
+	r3H1I3List = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: r3
+  namespace: istio-system
+spec:
+  actions:
+  - handler: h1.istio-system
+    instances:
+    - i3list
+---
+`
+
+	i4Quota = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: i4quota
```

Comment:
this should be "requestCount" or we change the "requestCount" below to "i4quota"